index.js

```diff
@@ -139,7 +188,43 @@ const sendMessage = async (data, topic) => {
     }
 };
 
+/**
+ * gets kafka topic metadata
+ * @param {Object} admin The admin object after calling connect
+ * @param {*} topic      The name of the topic
+ * @returns topic metadata
+ */
+const getTopicMetadata = async (admin, topic) => {
+    const metadata = await admin.fetchTopicMetadata({ topics: [topic] });
+
+    await admin.disconnect();
+
+    return metadata;
```

Comment:
Is there a reason why admin is disconnected after individual operation?
What if we need to get meta for multiple topics or do some other operation with the admin?
We should delegate the responsibility of the disconnecting the client to the code which created the admin client.
```suggestion
   return await admin.fetchTopicMetadata({ topics: [topic] });
```

Reply:
Yes, we don't want to keep an admin connection open, it should only be used for treating topics. It will not be used by any consumer.

Reply:
If the intent is to be restrictive to limit the admin connection to specific operations and disconnect them immediately, then the we should not expose the api to connect and use as admin outside such operations.

We should abstract the step to establish the connection within these functions.

Reply:
The api is exposed to make it more testable as, the objective is to have singular responsibility in functions

Reply:
Exposing the API for test is fine. But the downside is that it not restrictive and we don't have control over the admin operations that can be carried out.

Reply:
Since this function is doing more than fetching topic, can we rename the method to keep it explicit.
`fetchTopicMetadata` -> `fetchTopicMetadataAndDisconnect`
`createTopic` -> `createTopicAndDisconnect`

Reply:
I added disconnect in the finally block and that should allow the definition to remain same I think